### DISCUSSION TITLE
I can't seem to succeed in deploying KEYMAP_editor on-premises

### DISCUSSION THREAD

**Discussion by Deer-Xmj** [nickcoutsos/keymap-editor#250] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/f55c7092-c117-4027-9a3a-25e8a2f31cb8)
I followed the instructions step by step, and I have to mention that npm install has a high probability of failure and is very slow, is it impossible to successfully deploy locally now?

**Comment by nickcoutsos** [comment]
To be honest the public repo is very old and may not be that worthwhile to run locally unless you want to build something new on top of it.

I haven't tested it in a long time and I don't do any development on Windows so I don't know if that's the main issue. I did just give it a try with a fresh clone, npm install seemed to start up without error. Could you elaborate on npm being slow/unreliable?

**Comment by Deer-Xmj** [comment]
When I used the npm install command, I couldn't complete the download for dependencies smoothly, so I skipped this step, so I'll consider changing the download source tomorrow, because I think the big guy's keymap tool is good, but I'm worried that one day, you can't continue to support it, so I want to keep a copy on my local side, but the result seems to be terrible, and then I have the idea of taking a shortcut through git's workflow and then creating a controllable web page, if it's convenient for you, Can you give me some of the configuration files you used to configure this git workflow program, if it's too private, I won't force you, thank you for your reply

**Comment by Deer-Xmj** [comment]
![image](https://github.com/user-attachments/assets/2250dabf-0a4a-4ced-8166-8d8f67f4abd9)
My words may confuse you.,Actually, I want to skip the steps of relying on downloads by letting git compile.,But I can't seem to get the correct configuration file.,In addition, I want to keep a local program that you're now open to the public.,But I have a question that I don't know if this program is saved by this relatively outdated repository or if it's a newly developed program on your side and then works out to show it.,If it's not suitable for the outside world.,Even if I bother you, Because I just want to keep it up, so that I can make changes according to my own ideas in the future

**Reply by nickcoutsos** [reply, marked as answer]
I'm not completely sure I'm following, are you trying to use GitHub's CI process to download and bundle dependencies and publish an artifact that you can download to avoid issues with NPM locally? I guess that could work but you may have to configure the action to run on a Windows image (again, I've never tried this).

**Reply by Deer-Xmj** [reply]
Okay, I see, thanks for the reply
